### DISCUSSION TITLE
Async persist not repopulating store

### DISCUSSION THREAD

**Discussion by AndrewIsh** [pmndrs/zustand#2605] [Q&A] [answered]
I have a small store that I need to persist to an API. The saving and retrieval of the store to/from the API via `setItem` & `getItem` are working great. But, for some reason, the persisted value is never merged back into my store after retrieval. I'm not doing anything special as what I'm persisting is just a serialised version of the store, so I'm a bit puzzled. I've even tried manually merging, but if I add a `merge` to my `persist` config, it is never called.

Here is my store:
```JS
import { create } from 'zustand';
import {
  devtools,
  persist,
  createJSONStorage,
  type StateStorage
} from 'zustand/middleware';
import { getSetting, saveSetting } from '@/services/settings';
import { ZUSTAND_DASHBOARD_STORAGE_NAME } from '@/core/constants';
import { dashboardSlice } from './dashboardSlice';
import type { DashboardState } from './types';

const storage: StateStorage = {
  getItem: async (name) => await getSetting(name),
  setItem: async (name, value) =>  await saveSetting(name, value),
  removeItem: () => null
};

const useDashboardState = create<DashboardState>()(
  persist(
    devtools((...a) => ({
      ...dashboardSlice(...a)
    })),
    {
      name: ZUSTAND_DASHBOARD_STORAGE_NAME,
      storage: createJSONStorage(() => storage),
      skipHydration: true,
      partialize: (state) => state.layouts
    }
  )
);

export default useDashboardState;
```

As you can see, I am skipping hydration and calling it manually in my component, this is working as I can see the API call to retrieve the store, the problem seems to be that it isn't being merged once retreived.

I'm hoping I'm doing something ridiculous that is obvious to those more experienced in these things! Any help will be gratefully received!

**Comment by dbritto-dev** [comment, marked as answer]
@AndrewIsh I believe this part 

```
partialize: (state) => state.layouts
```

should be like this

```
partialize: (state) => ({ layouts: state.layouts })
```

**Reply by AndrewIsh** [reply]
@dbritto-dev Good spot, thank you! I made that change and that seems to have fixed the issue. Many thanks for your help 👍
